src/Features/CSharp/Portable/Completion/CompletionProviders/InternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,145 @@
+                    {
+                        return node.Parent as AttributeSyntax;
+                    }
+                }
+            }
+            return default(AttributeSyntax);
+        }
+
+        private static void AddAssemblyCompletionItems(CompletionContext context, CancellationToken cancellationToken)
+        {
+            var currentProject = context.Document.Project;
+            foreach (var p in context.Document.Project.Solution.Projects)
+            {
+                if (p == currentProject)
+                {
+                    continue;
+                }
+                var completionItem = CompletionItem.Create(displayText: p.AssemblyName,
+                    properties: ImmutableDictionary.Create<string, string>().Add(ProjectIdKey, p.Id.Id.ToString()));
+                context.AddItem(completionItem);
+            }
+        }
+
+        public override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey = default(char?), CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (item.Properties.TryGetValue(ProjectIdKey, out var projectId))
+            {
+                var assemblyName = item.DisplayText;
+                var project = document.Project.Solution.GetProject(ProjectId.CreateFromSerialized(new System.Guid(projectId)));
+                var publicKey = await GetPublicKeyOfProjectAsync(project, cancellationToken).ConfigureAwait(false);
+                if (!string.IsNullOrEmpty(publicKey))
+                {
+                    assemblyName += ", PublicKey=" + publicKey;
+                }
+                var tc = new TextChange(item.Span, assemblyName);
+                return CompletionChange.Create(tc);
+            }
+            Debug.Fail("Project can't be found by projectId.");
+            return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<string> GetPublicKeyOfProjectAsync(Project p, CancellationToken cancellationToken)
+        {
+            var compilationOptions = p.CompilationOptions;
+            if (compilationOptions == null ||
+                (string.IsNullOrEmpty(compilationOptions.CryptoKeyFile) && string.IsNullOrEmpty(compilationOptions.CryptoKeyContainer)))
+            {
+                return string.Empty;
+            }
+            var c = await p.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            if (c.Assembly?.Identity?.IsStrongName ?? false)
+            {
+                return GetPublicKeyAsHexString(c.Assembly.Identity.PublicKey);
+            }
+            return string.Empty;
+        }
+
+        private static string GetPublicKeyAsHexString(ImmutableArray<byte> publicKey)
+        {
+            var builder = SharedPools.Default<StringBuilder>().Allocate();
+            try
+            {
+                builder.Clear();
```

Comment:
We have PooledStringBuilder.  You can call ToStringAndFree on it to make this a lot simpler.